compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -5052,6 +5470,30 @@ private boolean isParameter(ParserRuleContext ctx) {
         }
     }
 
+    /**
+     * Get the shortest insert solution at the given parser rule context. If the parser rule is a terminal,
+     * then return the corresponding terminal token kind as the fix. If the parser rule is a production,
+     * then navigate the shortest path and get next available terminal token kind as the fix.
+     *
+     * @param ctx Parser rule context
+     * @return Shortest insert solution at the given parser rule
+     */
+    @Override
+    protected Solution getInsertSolution(ParserRuleContext ctx) {
+        SyntaxKind kind = getExpectedTokenKind(ctx);
+        if (kind != SyntaxKind.NONE) {
+            return new Solution(Action.INSERT, ctx, kind, ctx.toString());
+        }
+        
+        if (hasAlternativePaths(ctx)) {
+            ctx = getShortestAlternative(ctx);
+            return getInsertSolution(ctx);
+        }
+        
+        ctx = getNextRule(ctx, 1);
+        return getInsertSolution(ctx);
```

Comment:
Why do we need this? Can't we just do this as we did in `XMLParserErrorHandler`

Reply:
Resolving as per our offline discussion.